### DISCUSSION TITLE
Variables in Scratch CAN'T be found in the html file

### DISCUSSION THREAD

**Discussion by MA-master-sec** [TurboWarp/packager#743] [Q&A]
for example,I created a variable named `Scratch_var_1`
but then I can't find it in the output file...

So how could I find and set/change it?

**Comment by GarboMuffin** [comment]
The packager does not compile Scratch variables into `var Scratch_var_1 = 3` or anything like that

You can try `vm.runtime.getTargetForStage().lookupVariableByNameAndType("Scratch_var_1, "").value`

**Reply by MA-master-sec** [reply]
‘Cannot read properties of undefined (reading 'lookupVariableByNameAndType')’
so what can I write in the second ""

**Reply by GarboMuffin** [reply]
If the error occurred reading lookupVariableByNameAndType then getTargetForStage returned undefined, most likely because you ran the code before the project loaded. Variables do not exist until the project has loaded. Don't touch the `""` - that means you're looking up a variable. Replace that with `"list"` if you want a list instead.

**Reply by MA-master-sec** [reply]
Can I put it here:
![image](https://github.com/TurboWarp/packager/assets/126303227/57146322-4c88-4847-9752-b4b945631e1b)
